testsuite/integration/manualmode/src/test/java/org/wildfly/test/manual/elytron/seccontext/DynamicSSLContextTestCase.java

```diff
@@ -0,0 +1,340 @@
+        CoreUtils.createKeyMaterial(KEYSTORES2_DIR, JKS);
+    }
+
+    @Deployment(name = DEPLOYMENT, managed = false)
+    @TargetsContainer(CONTAINER)
+    public static Archive<?> deploy() {
+        final WebArchive war = ShrinkWrap.create(WebArchive.class, "archive" + ".war")
+                .addAsManifestResource(new StringAsset("Dependencies: org.jboss.as.controller-client,org.jboss.dmr\n"), "MANIFEST.MF")
+                .addPackage(TestSuiteEnvironment.class.getPackage());
+        return war;
+    }
+
+    @Test
+    @RunAsClient
+    @InSequence(1)
+    public void startAndConfigureContainerToUseDynamicSSLContextAsDefault() {
+        if (!serverController.isStarted(CONTAINER)) {
+            serverController.start(CONTAINER);
+        }
+        try {
+            ModelControllerClient modelControllerClient = TestSuiteEnvironment.getModelControllerClient();
+            ManagementClient managementClient = new ManagementClient(modelControllerClient, SERVER_ADDRESS, getManagementPort(), "remote+http");
+
+            addHTTPSListener(managementClient, KEYSTORES_DIR, SERVER_KEYSTORE1_FILENAME, SERVER_KEY_STORE1, SERVER_KEY_MANAGER1, SERVER_TRUSTSTORE1_FILENAME, SERVER_TRUST_STORE1, SERVER_TRUST_MANAGER1, SERVER_SSL_CONTEXT1, HTTP_LISTENER_1, SOCKET_BINDING_1, PORT1);
+            addHTTPSListener(managementClient, KEYSTORES2_DIR, SERVER_KEYSTORE1_FILENAME, SERVER_KEY_STORE2, SERVER_KEY_MANAGER2, SERVER_TRUSTSTORE1_FILENAME, SERVER_TRUST_STORE2, SERVER_TRUST_MANAGER2, SERVER_SSL_CONTEXT2, HTTP_LISTENER_2, SOCKET_BINDING_2, PORT2);
+
+            addSSLContext(managementClient, KEYSTORES_DIR, CLIENT_KEYSTORE1_FILENAME, CLIENT_KEY_STORE1, CLIENT_KEY_MANAGER1, CLIENT_TRUSTSTORE1_FILENAME, CLIENT_TRUST_STORE1, CLIENT_TRUST_MANAGER1, CLIENT_SSL_CONTEXT1);
+            addSSLContext(managementClient, KEYSTORES2_DIR, CLIENT_KEYSTORE1_FILENAME, CLIENT_KEY_STORE2, CLIENT_KEY_MANAGER2, CLIENT_TRUSTSTORE1_FILENAME, CLIENT_TRUST_STORE2, CLIENT_TRUST_MANAGER2, CLIENT_SSL_CONTEXT2);
+
+            addAuthenticationContext(managementClient);
+            configureDefaultSSLContext(managementClient);
+
+            reloadServer(modelControllerClient);
+            deployer.deploy(DEPLOYMENT);
+        } catch (Exception e) {
+            Assert.fail();
+        }
+    }
+
+    @Test
+    @InSequence(2)
+    @OperateOnDeployment(DEPLOYMENT)
+    public void clientRunningInsideContainerUsesDynamicSSLContextTest() throws Exception {
+        // default ssl context is configured to be dynamic in the server
+        SSLSocketFactory sf = SSLContext.getDefault().getSocketFactory();
+        // both handshakes will succeed even though these ports require different truststore and keystore
+        SSLSocket socket = (SSLSocket) sf.createSocket(SERVER_ADDRESS, PORT1);
+        socket.startHandshake();
+        socket = (SSLSocket) sf.createSocket(SERVER_ADDRESS, PORT2);
+        socket.startHandshake();
```

Comment:
This test case is a good smoke test. We could also add a test case that involves a connection from one server to another server. I think there should be some examples of this in `ElytronRemoteOutboundConnectionTestCase`.

Reply:
@fjuma Added